extensions/linear/src/api/getTeams.ts

```diff
@@ -15,26 +15,19 @@ export type TeamResult = Pick<
   organization: Pick<Organization, "logoUrl">;
 } & {
   activeCycle?: Pick<Cycle, "id" | "number">;
-} & {
-  projects: { nodes: Project["id"][] };
 };
 
-export async function getTeams() {
+export async function getTeams(query: string = "") {
   const { graphQLClient } = getLinearClient();
   const { data } = await graphQLClient.rawRequest<{ teams: { nodes: TeamResult[] } }, Record<string, unknown>>(
     `
-      query {
-        teams {
+      query($query: String!) {
```

Comment:
Hi, Linear developer here 👋🏼 

To make things faster, I'd use a query only if the response cursor indicates that the response is paged. In 95% of cases, workspaces have fewer than 50 teams, so we can just fetch once and avoid using a query

Reply:
For sure, in the latest revision, I have supported conditional typeahead for both users and teams. Typeahead will only be supported in case there is a next page available on initial loading. If no next page is available, user queries will just be used client-side to filter the first page's results (current behavior).

This prevents unnecessary calls/queries to linear APIs while also being memory-safe for raycast by not loading all resources.